src/controllers/music/songs.js

```diff
@@ -8,197 +7,207 @@ import * as userSettings from '../../scripts/settings/userSettings';
-                        IncludeItemTypes: 'Audio',
-                        Recursive: true,
-                        Fields: 'AudioInfo,ParentId',
-                        StartIndex: 0,
-                        ImageTypeLimit: 1,
-                        EnableImageTypes: 'Primary'
-                    }
-                };
-
-                if (userSettings.libraryPageSize() > 0) {
-                    pageData.query['Limit'] = userSettings.libraryPageSize();
+export default function (view, params, tabContent) {
+    function getPageData(context) {
+        const key = getSavedQueryKey(context);
+        let pageData = data[key];
+
+        if (!pageData) {
+            pageData = data[key] = {
+                query: {
+                    SortBy: 'Album,SortName',
+                    SortOrder: 'Ascending',
+                    IncludeItemTypes: 'Audio',
+                    Recursive: true,
+                    Fields: 'AudioInfo,ParentId',
+                    StartIndex: 0,
+                    ImageTypeLimit: 1,
+                    EnableImageTypes: 'Primary'
                 }
+            };
 
-                pageData.query.ParentId = params.topParentId;
-                libraryBrowser.loadSavedQueryValues(key, pageData.query);
+            if (userSettings.libraryPageSize() > 0) {
+                pageData.query['Limit'] = userSettings.libraryPageSize();
             }
 
-            return pageData;
+            pageData.query.ParentId = params.topParentId;
+            libraryBrowser.loadSavedQueryValues(key, pageData.query);
         }
 
-        function getQuery(context) {
-            return getPageData(context).query;
-        }
+        return pageData;
+    }
 
-        function getSavedQueryKey(context) {
-            if (!context.savedQueryKey) {
-                context.savedQueryKey = libraryBrowser.getSavedQueryKey('songs');
-            }
+    function getQuery(context) {
+        return getPageData(context).query;
+    }
 
-            return context.savedQueryKey;
+    function getSavedQueryKey(context) {
+        let savedQueryKey;
+        if (!context?.savedQueryKey) {
+            savedQueryKey = libraryBrowser.getSavedQueryKey('songs');
         }
 
-        function reloadItems(page) {
-            loading.show();
-            isLoading = true;
-            const query = getQuery(page);
-            ApiClient.getItems(Dashboard.getCurrentUserId(), query).then(function (result) {
-                function onNextPageClick() {
-                    if (isLoading) {
-                        return;
-                    }
-
-                    if (userSettings.libraryPageSize() > 0) {
-                        query.StartIndex += query.Limit;
-                    }
-                    reloadItems(tabContent);
-                }
-
-                function onPreviousPageClick() {
-                    if (isLoading) {
-                        return;
-                    }
+        return savedQueryKey;
+    }
```

Comment:
Are these changes necessary?

Reply:
I reverted it. When I started out, something around this threw an error but now it doesn't. Don't know what happened